tests/planned_maintenance_test.py

```diff
@@ -0,0 +1,183 @@
+from datetime import datetime, timedelta
+from unittest.mock import Mock
+
+import pytest
+
+from zino.planned_maintenance import PlannedMaintenances
+from zino.state import ZinoState
+from zino.statemodels import (
+    AlarmEvent,
+    DeviceMaintenance,
+    EventState,
+    Port,
+    PortStateEvent,
+    PortStateMaintenance,
+    ReachabilityEvent,
+)
+
+
+def test_should_start_with_no_planned_maintenances(pms):
+    assert len(pms) == 0
+
+
+def test_pm_should_be_gettable_by_id(pms, active_pm):
+    assert pms[active_pm.id] == active_pm
+
+
+class TestGetPlannedMaintenances:
+    def test_get_started_planned_maintenances(self, pms, active_pm, ended_pm, old_pm):
+        pms.last_run = datetime.now() - timedelta(hours=1)
+        recent_pm = pms.create_planned_maintenance(
+            start_time=datetime.now() - timedelta(minutes=1),
+            end_time=datetime.now() + timedelta(days=1),
+            pm_class=DeviceMaintenance,
+            match_type="str",
+            match_expression="hello",
+            match_device="device",
+        )
+        started_pms = pms.get_started_planned_maintenances(now=datetime.now())
+        assert recent_pm in started_pms
+        assert active_pm not in started_pms
+        assert ended_pm not in started_pms
+        assert old_pm not in started_pms
```

Comment:
This seems to pack a lot of different asserts into a single test.  It would be better if each concept was tested individually using tests with more specific names.  E.g. if a test called `test_started_planned_maintenances_should_not_contain_already_active_pms` fails, I know more about what's going on.

Reply:
fixed